app/models/miq_queue.rb

```diff
@@ -237,7 +237,12 @@ def self.put_or_update(find_options)
     #   it's unordered nature.
     where_scope = if conds.key?(:args)
                     args = YAML.dump conds.delete(:args)
-                    MiqQueue.where(conds).where(['args = ?', args])
```

Comment:
I agree, I hope we can find some way to make this work here and do a deeper refactor on a different PR.